app/services/log_event_searcher.rb

```diff
@@ -0,0 +1,62 @@
+class LogEventSearcher
+
+  ALLOWED_EVENTS = ["account.create", "account.update",
+                    "account_user.create", "account_user.delete",
+                    "user.create"].freeze
+
+  attr_accessor :start_date, :end_date, :events, :query
+
+  def initialize(start_date: nil, end_date: nil, events: [], query: nil)
+    @start_date = start_date
+    @end_date = end_date
+    @events = filter_events(events)
+    @query = query
+  end
+
+  def filter_events(events)
+    Array(events).select { |event| event.in?(ALLOWED_EVENTS) }
+  end
+
+  def search
+    result = LogEvent.all
+    result = result.merge(filter_date) if start_date || end_date
+    result = result.merge(filter_event) if events.present?
+    result = result.merge(filter_query) if query.present?
+    result
+  end
+
+  def dates
+    [(start_date || LogEvent.minimum(:event_time)), (end_date || Date.current)]
```

Comment:
I don't really like the `LogEvent.minimum(:event_time)` because of the extra database call, which once this table gets big will get slow, especially if event_time is not indexed. What if you introduced a `beginning_of_time = 10.years.ago`. Then, while it may be arbitrary, it at least is saying what it is meaning to be.